File: 1813-maximum-erasure-value/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        boolean[] seen = new boolean[10001]; // Constraints: 0 <= nums[i] <= 10000
        int i = 0, score = 0, maxScore = 0;

        for (int j = 0; j < nums.length; j++) {
            while (seen[nums[j]]) {
                score -= nums[i];
                seen[nums[i]] = false;
                i++;
            }
            seen[nums[j]] = true;
            score += nums[j];
            maxScore = Math.max(maxScore, score);
        }

        return maxScore;
    }
}

File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) 
    {
        bool seen[10001];
        int i = 0, n = nums.size();
        int score = 0, maxscore = 0;
        for(int j = 0; j < n; j++)
        {
            while(seen[nums[j]])
            {
                score -= nums[i];
                seen[nums[i++]] = false;
            }
            seen[nums[j]] = true;
            score += nums[j];
            maxscore = max(score, maxscore);
        }
        return maxscore;
    }
};